### ISSUE TITLE
Add an easier way to disable select BuildTasks

### ISSUE THREAD

**Issue by kevingroeger** [silverstripe/silverstripe-framework#10799]
## Issue

For safety reasons, I would like to have an option to disable select BuildTasks. Currently you have to build an injector to do it. Which IMO isn't optimal. 

There is ofc, the DevConfirmationController which stops you from just mis-clicking a task that you don't want to run. But I would rather not even have them show up at all.

### Through config 

Add a way to configure the tasks via a config file. And change the isEnabled method.

Example:

```yml
---
Name: DevTasks
---
MyNameSpace\Dev\MyTask:
    enabled: false
```
```php
   /**
     * @return bool
     */
    public function isEnabled()
    {
        return $this->config()->get('enabled');
    }
```

------------

Are there any Security concerns that I am overseeing? Or can this just be implemented like suggested? Please let me know, I'll create a PR if this seems good 👍 

### PRs
- https://github.com/silverstripe/silverstripe-framework/pull/10801
- https://github.com/silverstripe/silverstripe-framework/pull/10804
- https://github.com/silverstripe/developer-docs/pull/281

**Comment by michalkleiner** [comment]
We would probably need to keep the `protected $enabled` as changing that could be breaking someone's tasks that extend from the core tasks. So it would be a BC breakage and should be done in a major version, unless we find a way how to make it as an enhancement and then it could go into the next minor.

**Comment by emteknetnz** [comment]
Yes it should be configurable, we can do this in a non-bc way.

```
abstract class BuildTask
{
    // change this to `bool` for CMS6
    private static ?bool $is_enabled = null;
    
    // @deprecated - remove in CM6
    protected $enabled = true;
    
    public function isEnabled(): bool
    {
        $isEnabled = $this->config()->get('enabled');
        if ($isEnabled === null) {
            Deprecation::warning('Null values not allowed in CMS6 ...');
            $isEnabled = static::$enabled;
        }
        return $isEnabled;
    }
```

**Comment by kevingroeger** [comment]
Created the PR: https://github.com/silverstripe/silverstripe-framework/pull/10801

Thanks for the suggestions!

**Comment by GuySartorelli** [comment]
PRs merged. This feature will be available from 5.1.0
